src/rust/src/lib.rs

```diff
@@ -770,6 +775,23 @@ pub(crate) fn compute_write_fee_per_1kb(
     ))
 }
 
+pub fn json_to_config_upgrade_set(file_name: &CxxString) -> Result<RustBuf, Box<dyn std::error::Error>>
```

Comment:
An alternative to taking JSON directly with this code is to use https://github.com/stellar/rs-stellar-xdr/pull/311 and just take XDR as input in core.